global/scripts/FieldMapPreprocessingAll.sh

```diff
@@ -42,14 +58,15 @@ opts_AddMandatory '--ofmapmagbrain' 'MagnitudeBrainOutput' 'image' "output disto
 
 opts_AddMandatory '--ofmap' 'FieldMapOutput' 'image' "output distortion corrected fieldmap image (rad/s)"
 
-#Optional Arguments
-opts_AddOptional '--echodiff' 'DeltaTE' 'number (milliseconds)' "echo time difference for Siemens and Philips fieldmap images (in milliseconds)"
+# options --echodiff is now mandatory; used by all "--method" options (and necessary for processing of all vendors). 
+opts_AddMandatory '--echodiff' 'DeltaTE' 'number (milliseconds)' "echo time difference for fieldmap images (in milliseconds)"
 
-opts_AddOptional '--fmap' 'GEB0InputName' 'number (degrees)' "input General Electric fieldmap with fieldmap in deg and magnitude image"
+# Optional Arguments
+opts_AddOptional '--fmapgelegacy' 'GEB0InputName' 'image (hertz and magnitude)' "GE HealthCare Legacy fieldmap with field map in Hz and magnitude image included as two volumes in a single file"
```

Comment:
```suggestion
opts_AddOptional '--fmapcombined' 'GEB0InputName' 'image (hertz and magnitude)' "GE HealthCare Legacy fieldmap with field map in Hz and magnitude image included as two volumes in a single file" '' '--fmap'
```

This is how to support a (hidden) legacy flag with our option parsing library.  The variable name could also be made more generic, if you feel like it.

Reply:
integrated in commit [755e64c](https://github.com/Washington-University/HCPpipelines/pull/285/commits/755e64c33bde815fd2aa16f7cd62cb3af79f8c91)